packages/phpstan-rules/src/DeadCode/UnusedPublicClassConstRule.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\PHPStanRules\DeadCode;
+
+use Nette\Utils\Arrays;
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Node\CollectedDataNode;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleError;
+use PHPStan\Rules\RuleErrorBuilder;
+use Symplify\PHPStanRules\Collector\ClassConstFetchCollector;
+use Symplify\PHPStanRules\Collector\PublicClassLikeConstCollector;
+use Symplify\RuleDocGenerator\Contract\DocumentedRuleInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Symplify\PHPStanRules\Tests\Rules\DeadCode\UnusedPublicClassConstRule\UnusedPublicClassConstRuleTest
+ */
+final class UnusedPublicClassConstRule implements Rule, DocumentedRuleInterface
+{
+    /**
+     * @var string
+     */
+    public const ERROR_MESSAGE = 'Class constant "%s" is never used';
```

Comment:
IMHO it should be less enigmatic (more descriptive), but maybe there's such convention 😉

Reply:
This comment relates to `packages/phpstan-rules/docs/rules_overview.md` basically, because as an error message it's totally fine. So most probably rules overview's generator could be improved to use some kind of description instead of error message used for reporting.

Reply:
> So most probably rules overview's generator could be improved to use some kind of description instead of error message used for reporting.

That's by convention, to avoid duplicating same meaning. Having differente content would be confusing. 

But improving this message would make docs more clear too :+1:  Got an idea on how to improve this one?